### ISSUE TITLE
[proposal] google_compute_resource_policy warn policy name follows regex rule

### ISSUE THREAD

**Issue by jlosito** [terraform-linters/tflint-ruleset-google#66]
I have a module which looks like the following.

```
resource "google_compute_resource_policy" "snapshot_policy" {
  name   = "snapshot_policy"
  region = var.gcp_region
  snapshot_schedule_policy {
    schedule {
      hourly_schedule {
        hours_in_cycle = 12
        start_time     = "03:00"
      }
    }
    retention_policy {
      max_retention_days = 61
    }
  }
}
```

When I try to provision the snapshot policy, I get the following error because the name of the policy does not follow the convention even though a `terraform plan` does not indicate any error before the `apply`. It appears that hyphens are prefered rather than underscores in names.

```
Error: Error creating ResourcePolicy: googleapi: Error 400: Invalid value for field 'resource.name': 'snapshot_policy'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)', invalid
```

I believe a `tflint` using the google ruleset does not catch this either.

I'm currently using the latest version of the ruleset from this project and below are the versions of tflint and go I have.
```
% go version
go version go1.15.6 darwin/amd64

% tflint --version
TFLint version 0.23.0
+ ruleset.google (0.4.0)
```
